Add basic docs for serializers (#679)
software-mansion/starknet.py: docs/guide/serialization.rst

Serialization
#############

Data serialization
-------------------

> Starknet.py transforms python values to Cairo values and the other way around.

Reviewer: ```suggestion
Starknet.py **serializes** python values to Cairo values and **deserializes** Cairo values to python values.
```

   * - tuple
     - tuple
   * - named tuple

>      - DataclassTuple

Reviewer 2: Same here
